pkg/ipam/ipam.go

```diff
@@ -246,12 +246,16 @@ func AllocateIP(slsSubnet sls_common.IPV4Subnet, xname xnames.Xname, alias strin
 
 	// Verify this switch is unique within the subnet
 	for _, ipReservation := range slsSubnet.IPReservations {
-		if ipReservation.Name == alias {
-			return sls_common.IPReservation{}, fmt.Errorf("ip reservation with name (%v) already exits", alias)
-		}
-
-		if ipReservation.Comment == xname.String() {
-			return sls_common.IPReservation{}, fmt.Errorf("ip reservation with xname (%v) already exits", xname.String())
+		matchingAlias := ipReservation.Name == alias
+		matchingXName := ipReservation.Comment == xname.String()
+
+		if matchingAlias && matchingXName {
+			// IP reservation already exists
+			return sls_common.IPReservation{}, nil
+		} else if ipReservation.Name == alias {
```

Comment:
Could use the matchingAlias variable, and the next else if could use matchingXName

Reply:
Good call Shane! That was my intent, and looks like I didn't fall through with it.